File: Problem-Solveing/Leetcode/DSA/GoLang/374.go
// 374. Guess Number Higher or Lower
// https: //leetcode.com/problems/guess-number-higher-or-lower/submissions/
// Runtime: 2 ms, faster than 33.96% of Go online submissions for Guess Number Higher or Lower.
// Memory Usage: 1.9 MB, less than 99.37% of Go online submissions for Guess Number Higher or Lower.

// func guessNumber(n int) int {
// 	l, h := 1, n
// 	for l < h {
// 		mid := (l + h) / 2
// 		if guess(mid) == 1 {
// 			l = mid + 1
// 		} else {
// 			h = mid
// 		}
// 	}
// 	return l
// }

File: Problem-Solveing/Leetcode/DSA/Python/374.py
# https://leetcode.com/problems/guess-number-higher-or-lower/
# 374. Guess Number Higher or Lower

### My solution
# class Solution:
#     def guessNumber(self, n: int) -> int:
#         l=1
#         h=n
#         while l<=h:
#             m = (h+l)//2
#             if guess(m) == 0:
#                 return m
#             elif guess(m) > 0:
#                 l = m +1
#             else:
#                 h = m - 1
#         return m    

### Best Solution
# class Solution:
#     def guessNumber(self, n: int) -> int:
#         l, h = 1, n
#         while l < h:
#             mid = (l+h)//2
#             if guess(mid) == 1:
#                 l = mid+1
#             else:
#                 h = mid
#         return l